tests/devices/unit_tests/test_undulator.py

```diff
@@ -95,6 +108,27 @@ async def test_length_not_propagated_if_not_supplied():
             "UND-01",
             name="undulator",
             poles=80,
+            id_gap_lookup_table_path=ID_GAP_LOOKUP_TABLE_PATH,
         )
     assert undulator.length is None
     assert "undulator-length" not in (await undulator.read_configuration())
+
+
+@pytest.mark.parametrize(
+    "dcm_energy, expected_output", [(5730, 5.4606), (7200, 6.045), (9000, 6.404)]
+)
+def test_correct_closest_distance_to_energy_from_table(dcm_energy, expected_output):
+    energy_to_distance_table = np.array([[5700, 5.4606], [7000, 6.045], [9700, 6.404]])
+    assert (
+        _get_closest_gap_for_energy(dcm_energy, energy_to_distance_table)
+        == expected_output
+    )
+
+
+async def test_when_gap_access_is_disabled_set_energy_then_error_is_raised(
+    undulator,
+):
+    set_mock_value(undulator.gap_access, UndulatorGapAccess.DISABLED)
+    with pytest.raises(AccessError):
+        # AccessError("Undulator gap access is disabled. Contact Control Room")
```

Comment:
Nit: What does this comment give us?

Reply:
I left it temporarily for reference to show when the test wasn't passing. it passes now, so I deleted it